addons/common/fnc_runQTE.sqf

```diff
@@ -0,0 +1,97 @@
+#include "script_component.hpp"
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_runQTE
+
+Description:
+	Runs a Quick time Event.
+
+Parameters:
+	_object - <OBJECT>
+	_args - Extra arguments passed to the _on... functions<ARRAY>
+ 	_onDisplay - Code callback on displayable event passed [_args, _qte_sequence, _qte_history]. <CODE, STRING>
+	_onFinish - Code callback on Quick-Time Event completed passed [_args, _elapsedTime]. <CODE, STRING>
+	_onFinish - Code callback on Quick-Time Event timeout/outranged passed [_args, _elapsedTime]. <CODE, STRING>
+ 	_qte_sequence - Quick-Time sequence made up of ["↑", "↓", "→", "←"] <ARRAY>
+ 	_max_distance - max interaction distance from attached object <NUMBER> (default: 10) 
+ 	_timeout - ingame timeout <NUMBER> (default: 30)
+
+Example:
+    [car,
+	[], 
+	{ 
+	hint format [
+		"%1 \n %2",
+		[_this select 1] call CBA_fnc_getFormattedQTESequence,
+		[_this select 2] call CBA_fnc_getFormattedQTESequence
+		]
+	}, 
+	{ 
+	hint "Finished!"; 
+	},
+	{ 
+		hint "Failure!"; 
+	},
+	["↑", "↓", "→", "←"]] call CBA_fnc_runQTE
+
+Returns:
+    Nil
+
+Author:
+    john681611
+---------------------------------------------------------------------------- */
+
+
+params ["_object", "_args", "_onDisplay", "_onFinish", "_onFail", "_qte_sequence", ["_max_distance", 10], ["_timeout", 30]];
+if (GVAR(QTERunning)) exitWith {
+	TRACE_1("QTE already running qeueing up",GVAR(QTERunning));
+	[{
+		!GVAR(QTERunning)
+	}, {
+		_this call FUNC(runQTE);
+	}, _this] call CBA_fnc_waitUntilAndExecute;
```

Comment:
I'm still not a fan of this. Depending, we might have to wait indefinitely. It also means that multiple QTE can queue up, which *could* cause unforeseen headaches, albeit unlikely.

I think we'd be better off returning a boolean telling the caller if the QTE was successfully run or not.